### ISSUE TITLE
Setting the chips in ISP mode

### ISSUE THREAD

**Issue by mid-kid** [frank-zago/isp55e0#7]
The readme contains the following mention: "When set in ISP mode".  
This implies that the devices have an ISP mode that can be enabled.
I have a device containing a CH582M, with exposed debug port, and I've been scrolling through the datasheets, but I cannot find any documentation on how this ISP mode works, nor how to enable it.
Is it a pin that must be held high or something? Do you know where I can find any information about this?

**Comment by frank-zago** [comment]
Two pins must be shorted during power up. Which ones depend on the board and chip you have.
It could be a button that should be pressed, or a header that needs a jumper, ... Some of my breakout boards have a "prog" or "DL" label to indicate which ones.

**Comment by mid-kid** [comment]
Thanks for the reply!
I see, my board doesn't have an exposed pin for this, seemingly, but I've found an evaluation board for the exact chip, with a "download" key, do you reckon it's this one?
![](https://gd4.alicdn.com/imgextra/i4/926072809/O1CN01KsAQAu1WcZI9uqbq3_!!926072809.png)

Do you know of any datasheets that document this pin? It would help if I knew what term they use to describe this pin, or could get more information on the bootloader in general.

**Comment by frank-zago** [comment]
Yes, "download" would make sense.

I have no information about that chip. Usually there are no reserved pin for that purpose; the pin(s) are just multiplexed in their usage. 
In that image above, the download switch could be connecting USB VBUS to pin 28. Maybe. It's hard to follow. Doing it blindly on your board could fry the chip.

It's also possible that your board doesn't have a bootloader programmed, which is why there is no programming switch.

**Comment by mid-kid** [comment]
Alright, thanks a lot for your answer. I'll mess around with my board and see what I can find.

**Comment by nikitos1550** [comment]
As I remember PB22 should be tied to VCC on power up (not unreset, but only power up) in order to run bootloader mode.

**Comment by frank-zago** [comment]
You can find a schematic for a board there: http://152.32.187.208:8080/yd-data/YD-CH582M/. Boot is shorting PB22 to ground AFAICT.
